packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,545 @@
+
+        emit TransactionExecuted(callingSafe, nonce, txHash);
+    }
+
+    /// @notice Called by the Safe after executing a transaction
+    /// @dev Implementation of IGuard interface
+    function checkAfterExecution(bytes32, bool) external override {
+        // Do nothing
+        // In order to follow the Checks-Effects-Interactions pattern,
+        // all logic should be done in the checkTransaction function.
+    }
+
+    ////////////////////////////////////////////////////////////////
+    //              Internal State-Changing Functions             //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Increase the cancellation threshold for a safe
+    /// @dev This function must be called only once and only when calling cancel
+    /// @param _safe The Safe address to increase the cancellation threshold for.
+    function _increaseCancellationThreshold(Safe _safe) internal {
+        SafeState storage safeState = _safeState[_safe];
+
+        if (safeState.cancellationThreshold < maxCancellationThreshold(_safe)) {
+            uint256 oldThreshold = safeState.cancellationThreshold;
+            safeState.cancellationThreshold++;
+            emit CancellationThresholdUpdated(_safe, oldThreshold, safeState.cancellationThreshold);
+        }
+    }
+
+    /// @notice Reset the cancellation threshold for a safe
+    /// @dev This function must be called only once and only when calling checkAfterExecution
+    /// @param _safe The Safe address to reset the cancellation threshold for.
+    function _resetCancellationThreshold(Safe _safe) internal {
+        SafeState storage safeState = _safeState[_safe];
+        uint256 oldThreshold = safeState.cancellationThreshold;
+        safeState.cancellationThreshold = 1;
+        emit CancellationThresholdUpdated(_safe, oldThreshold, 1);
+    }
+
+    ////////////////////////////////////////////////////////////////
+    //              External State-Changing Functions             //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Configure the contract as a timelock guard by setting the timelock delay
+    /// @dev This function is only callable by the Safe itself, and will revert if the guard is not enabled on the Safe.
+    ///      Requiring a call from the Safe itself (rather than accepting signatures directly as in cancelTransaction())
+    ///      is important to ensure that maliciously gathered signatures will not be able to instantly reconfigure
+    ///      the delay to zero.
+    /// @param _timelockDelay The timelock delay in seconds (0 to clear configuration)
+    /// @dev We considered several implementations to allow for a pause mechanism of sorts, which could be used if the
+    /// on-call team needed more time. Ultimately, we rejected all of them in favour of better on-call processes.
+    function configureTimelockGuard(uint256 _timelockDelay) external {
```

Comment:
Does safe expose its version onchain? If so we should check the 1.3.0 compatibility restriction in this method, otherwise it's possible to brick your safe by installing this on a <1.3.0 safe because `scheduleTransaction` will revert every time it tries to call `checkSignatures`